core/blockdag/blockdag.go

```diff
@@ -351,10 +379,35 @@ func (bd *BlockDAG) GetBlock(h *hash.Hash) IBlock {
 // Acquire one block by hash
 // Be careful, this is inefficient and cannot be called frequently
 func (bd *BlockDAG) getBlock(h *hash.Hash) IBlock {
+	return bd.getBlockAdvanced(h, nil)
+}
+
+func (bd *BlockDAG) getBlockAdvanced(h *hash.Hash, tx database.Tx) IBlock {
```

Comment:
- looks `getBlockAdvanced` only called by `getblock` , it means `tx != nil` clause never executed
- the `if-else` clause surrounded by the duplicated logic of calling of `DBGetBlockIdByHash`, its better to refactor the duplicated logic itself into a func alone and eliminate the requisite for the `if-else` clause.

Reply:
62dda74